pkg/kubelet/kubelet_node_status.go

```diff
@@ -341,23 +341,32 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 }
 
 func (kl *Kubelet) setNodeStatusMachineInfo(node *api.Node) {
+	// Note: avoid blindly overwriting the capacity in case opaque
+	//       resources are being advertised.
+	if node.Status.Capacity == nil {
+		node.Status.Capacity = api.ResourceList{}
```

Comment:
the set call looks fine here, but I think it may be still possible that Kubelet will have node capacity cached elsewhere and will never observe the opaque resources were added.   I am not sure if that could result in failed admission checks.  I will need to vet that part of the code when not on phone